students/JerryH/Lesson04/mailroom2.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+from collections import defaultdict
+
+# List of sample donors and donations
+donors = defaultdict(list, {
+    "Bill Gates": [234.22, 45645.24, 43953.09, 98823],
+    'Jeff Bezo': [4564.23],
+    'Mike Dell': [299.09, 26273.67],
+    'Harry Potter': [8234.09, 83948.04, 7834.23],
+    'Ben Williams': [29283.00, 1334.34],
+    'Guy James': [93.00, 34.34]
+    })
+
+menu = {
+    'op1': "Send a Thank You",
+    'op2': "Create a Report",
+    'op3': "Send Letters To Everyone",
+    'op4': "Quit"
+    }
+
+QUIT_OPT = '4'
+
+def prompt():
+    return input("Please choose the following options:\n1) {op1}.\n2) {op2}.\n3) {op3}.\n4) {op4}\n".format(**menu))
+
+
+def donor_name_prompt():
+    return input("Send a Thank You - Please enter a full name or type \"list\""
+        "to list the current donors:\n")
+
+
+def get_all_donor_names():
+    return list(donors)
+
+
+def list_all_donor_names():
+    for each_name in sorted(get_all_donor_names()):
+        print("{}".format(each_name))
+
+
+def send_thank_you():
+    donor_name = None
+    while not donor_name:
+        donor_name = donor_name_prompt()
+        if donor_name.lower() == "list":
+            donor_name = None
+            list_all_donor_names()
+
+    donation = float(input("Please enter the donation amount:\n"))
+
+    # If the donnor doesn't exist in the donor dictionary - add his info
+    if donor_name not in get_all_donor_names():
+        # print(donors)
+        donors[donor_name] = [donation]
+    # else append the donation into the existing donors
+    else:
+        donors[donor_name].append(donation)
```

Comment:
plz review my previous comment on this area.

what you have here is correct and will work, but we can make it better 🙂 

since you are using defaultdict now, you do not need to check if donor exists or not, you can just call append on the new donor name. it makes things much simpler.

I am happy to explain this more extensively if you would like me to, let me know.